From 479415c78ebd378a6cf37e40a6f707840af13706 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Thu, 15 Jun 2017 13:19:34 +0800
Subject: [PATCH 1893/5242] MLK-15063: ASoC: fsl: add dpcm_merged_chan for
 machine driver

commit  bd136bf3bbba82d08e64f050dd2e0c86438ad13d from
https://source.codeaurora.org/external/imx/linux-imx.git

Same as commit cfe36e2e7fce ("MLK-15043-2: ASoC: imx-cs42888: fix
noise issue with FE-BE case"). need to add same configuration
for imx-wm8960, imx-wm8962, imx-mqs.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit 7195ad8ff56857028a6990296cc06e1c08a4e48e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-mqs.c    |    1 +
 sound/soc/fsl/imx-wm8960.c |    1 +
 sound/soc/fsl/imx-wm8962.c |    1 +
 3 files changed, 3 insertions(+)

diff --git a/sound/soc/fsl/imx-mqs.c b/sound/soc/fsl/imx-mqs.c
index a09a52a..a393635 100644
--- a/sound/soc/fsl/imx-mqs.c
+++ b/sound/soc/fsl/imx-mqs.c
@@ -120,6 +120,7 @@ static int be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.ignore_pmdown_time = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 0,
+		.dpcm_merged_chan = 1,
 	},
 	{
 		.name = "HiFi-ASRC-BE",
diff --git a/sound/soc/fsl/imx-wm8960.c b/sound/soc/fsl/imx-wm8960.c
index bff066f..9fbd86b 100644
--- a/sound/soc/fsl/imx-wm8960.c
+++ b/sound/soc/fsl/imx-wm8960.c
@@ -405,6 +405,7 @@ static int be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.ignore_pmdown_time = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
+		.dpcm_merged_chan = 1,
 	},
 	{
 		.name = "HiFi-ASRC-BE",
diff --git a/sound/soc/fsl/imx-wm8962.c b/sound/soc/fsl/imx-wm8962.c
index a9d3a91..308736f 100644
--- a/sound/soc/fsl/imx-wm8962.c
+++ b/sound/soc/fsl/imx-wm8962.c
@@ -712,6 +712,7 @@ static int imx_wm8962_probe(struct platform_device *pdev)
 		data->dai[1].ignore_pmdown_time = 1;
 		data->dai[1].dpcm_playback = 1;
 		data->dai[1].dpcm_capture = 1;
+		data->dai[1].dpcm_merged_chan = 1;
 
 		data->dai[2].name = "HiFi-ASRC-BE";
 		data->dai[2].stream_name = "HiFi-ASRC-BE";
-- 
1.7.9.5

